vms/avm/service.go

```diff
@@ -665,14 +665,26 @@ func (service *Service) ImportKey(r *http.Request, args *ImportKeyArgs, reply *I
 		return fmt.Errorf("problem saving key %w", err)
 	}
 
-	addresses, _ := user.Addresses(db)
-	addresses = append(addresses, sk.PublicKey().Address())
+	addresses, err := user.Addresses(db)
+	if err != nil {
+		return fmt.Errorf("problem saving key while getting existing addresses: %w", err)
+	}
+	newAddress := sk.PublicKey().Address()
+	exists := false
+	for _, address := range addresses {
+		if newAddress.Equals(address) {
+			exists = true
+		}
+	}
+	if !exists {
+		addresses = append(addresses, newAddress)
 
-	if err := user.SetAddresses(db, addresses); err != nil {
-		return fmt.Errorf("problem saving addresses: %w", err)
+		if err := user.SetAddresses(db, addresses); err != nil {
+			return fmt.Errorf("problem saving addresses: %w", err)
+		}
 	}
 
-	reply.Address = service.vm.Format(sk.PublicKey().Address().Bytes())
+	reply.Address = service.vm.Format(newAddress.Bytes())
```

Comment:
If we move
```
reply.Address = service.vm.Format(newAddress.Bytes())
```
 up to where we define `newAddress`, then we could just `return nil` inside the duplication check, and then we'll have removed the need for the `exists` variable.

Reply:
Fixed